### ISSUE TITLE
Support `keepLast` retention in k8up

### ISSUE THREAD

**Issue by seanhamlin** [uselagoon/lagoon#2962]
k8up supports a retention key `keepLast` https://k8up.io/k8up/1.2/how-tos/schedules.html

If would be nice if there was a way to pass these args into the KBD to then pass into the k8up schedule object.

Related to https://github.com/amazeeio/lagoon-kbd/issues/92

**Comment by Schnitzel** [comment]
can you elaborate the idea behind this? also we could make this the default? then we don't have to add additional arguments to KBD, helmcharts, etc/

**Comment by seanhamlin** [comment]
Yeah sure, so the idea is we have a customer that is looking to create 6 hourly backups. The current retention keys are a little confusing.

We currently can use

```
keepHourly: 72
```

Where I would prefer to have 

```
keepLast: 12
```

To keep the last 3 days worth of 6 hourly backups.

> also we could make this the default?

Potentially, likely it would still need to be combo'ed with `keepMonthly` etc if longer term backups are needed.

**Comment by Schnitzel** [comment]
mhh I'm not sure I understand, so if you say `create 6 hourly backups` you mean a backup every 6 hours? or do backups every hour and want to keep the last 6 hourly ones?

so assuming you do a backup every 6 hours and you want to keep them the last 3 days (4 backups per day), then `keepHourly: 72` would should work, as the documentation (https://restic.readthedocs.io/en/latest/060_forget.html) says:

> `--keep-hourly n` for the last `n` hours in which a snapshot was made, keep only the last snapshot for each hour.

we don't have 72 backups but based on how I read this this would keep hourly backups it could find in the last 72h, which would be all backups that have been created.

So I'm not sure why we need to add the `keepLast` feature as it currently can be implemented with what we have?

**Comment by seanhamlin** [comment]
> you mean a backup every 6 hours? 

Yes I do.

And yes this is achievable with the current functionality, but it did take a lot of reading of restic and k8up docs to discern this. If you feel this is not confusing, then feel free to close this issue.

**Comment by Schnitzel** [comment]
I'm more concerned adding a new functionality to a system that already works, I would rather fix actual bugs or implement features that don't work  yet :)

But I let @tobybellwood decide on this
